### PR TITLE
[PLT-1471] Remove pydantic v2 deprecate warnings

### PR DESCRIPTION
**Pull request by Gabefire**
# Description

* Removed pydantic warnings from SDK. 
* Removed datetime.utcnow from test which is also deprecated

### CODE REVIEW

File: libs/labelbox/src/labelbox/data/annotation_types/types.py [hunk 1]
```diff
@@ -5,19 +5,22 @@
...
 
 DType = TypeVar("DType")
 DShape = TypeVar("DShape")
 
 
 class _TypedArray(np.ndarray, Generic[DType, DShape]):
     @classmethod
-    def __get_validators__(cls):
-        yield cls.validate
+    def __get_pydantic_core_schema__(
```
Comment (Gabefire) [inline libs/labelbox/src/labelbox/data/annotation_types/types.py:17]: I have a story to simplify this class

File: libs/labelbox/src/labelbox/schema/data_row_metadata.py [hunk 2]
```diff
@@ -49,7 +56,7 @@ class DataRowMetadataKind(Enum):
 # Metadata schema
 class DataRowMetadataSchema(BaseModel):
     uid: SchemaId
-    name: str = Field(strip_whitespace=True, min_length=1, max_length=100)
```
Comment (Gabefire) [inline libs/labelbox/src/labelbox/schema/data_row_metadata.py:52]: strip_whitespace is depercated in pydantic V2 and gives warnings to users

File: libs/labelbox/src/labelbox/schema/data_row_metadata.py [hunk 3]
```diff
@@ -36,6 +37,12 @@
     format_iso_from_string,
 )
 
+Name = Annotated[
```
Comment (vbrodsky) [inline libs/labelbox/src/labelbox/schema/data_row_metadata.py:40]: good

**Review by vbrodsky — APPROVED** [review]
Can we also remove some more warnings re 
DeprecationWarning: datetime.datetime.utcnow() 

Also can you squash these 53 commits?
Also can you add to the end of the epic to make sure we remove / do not have sdk deprecation warnings

**Pull request merged**